[Bug]: Not able to install it using GitHub address in Node 16

### Version

28.0.8

### Steps to reproduce

Install it using GitHub repository address git+https://github.com/kulshekhar/ts-jest.git
Try to run `npm ci` in Github Actions with NodeJS version 16.

### Expected behavior

ts-jest to be built and installed.

### Actual behavior

It will fail to install ts-jest with this error:

```
npm ERR! /home/runner/.npm/_cacache/tmp/git-clonevJa6QI/node_modules/execa/index.js:212
[26](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:27)
npm ERR! 		throw error;
[27](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:28)
npm ERR! 		^
[28](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:29)
npm ERR! 
[29](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:30)
npm ERR! Error: Command failed with exit code 128: git diff-index --name-only HEAD
[30](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:31)
npm ERR! fatal: not a git repository (or any of the parent directories): .git
[31](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:32)
npm ERR!     at makeError (/home/runner/.npm/_cacache/tmp/git-clonevJa6QI/node_modules/execa/lib/error.js:60:11)
[32](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:33)
npm ERR!     at Function.module.exports.sync (/home/runner/.npm/_cacache/tmp/git-clonevJa6QI/node_modules/execa/index.js:194:17)
[33](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:34)
npm ERR!     at Object.<anonymous> (/home/runner/.npm/_cacache/tmp/git-clonevJa6QI/scripts/post-build.js:5:11)
[34](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:35)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
[35](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:36)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
[36](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:37)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
[37](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:38)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
[38](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:39)
npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
[39](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:40)
npm ERR!     at node:internal/main/run_main_module:17:47 {
[40](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:41)
npm ERR!   shortMessage: 'Command failed with exit code 128: git diff-index --name-only HEAD',
[41](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:42)
npm ERR!   command: 'git diff-index --name-only HEAD',
[42](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:43)
npm ERR!   escapedCommand: 'git diff-index --name-only HEAD',
[43](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:44)
npm ERR!   exitCode: 128,
[44](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:45)
npm ERR!   signal: undefined,
[45](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:46)
npm ERR!   signalDescription: undefined,
[46](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:47)
npm ERR!   stdout: '',
[47](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:48)
npm ERR!   stderr: 'fatal: not a git repository (or any of the parent directories): .git',
[48](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:49)
npm ERR!   failed: true,
[49](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:50)
npm ERR!   timedOut: false,
[50](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:51)
npm ERR!   isCanceled: false,
[51](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:52)
npm ERR!   killed: false
[52](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:53)
npm ERR! }
[53](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:54)
npm ERR! npm ERR! code 1
[54](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:55)
npm ERR! npm ERR! path /home/runner/.npm/_cacache/tmp/git-clonevJa6QI
[55](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:56)
npm ERR! npm ERR! command failed
[56](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:57)
npm ERR! npm ERR! command sh -c npm run build
[57](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:58)
npm ERR! 
[58](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:59)
npm ERR! npm ERR! A complete log of this run can be found in:
[59](https://github.com/sassanh/template-to-pdf/runs/8051920469?check_suite_focus=true#step:4:60)
npm ERR! npm ERR!     /home/runner/.npm/_
```

### Debug log

Not applicable

### Additional context

I'm doing this because this pull request is not released yet: https://github.com/kulshekhar/ts-jest/pull/3767
This is the line that's causing the issue: https://github.com/kulshekhar/ts-jest/blob/main/scripts/post-build.js#L5
I think in NPM associated with Node 16 `.git` is removed the repository is cloned. Is there any other way we can have that condition without running git?

### Environment

```shell
System: Github Actions infrastructure, ubuntu-latest
Node: 16
```

--- a/scripts/post-build.js
+++ b/scripts/post-build.js
@@ -2,10 +2,14 @@ const { computePackageDigest } = require('./lib/bundle')
 const execa = require('execa')
 const { generatedPath, rawCompilerOptionsFileName } = require('./lib/paths')
 
-if (execa.sync('git', ['diff-index', '--name-only', 'HEAD']).stdout.includes(rawCompilerOptionsFileName)) {
-  throw new Error(
-    `Tsconfig options have changed. Please check the modified generated ${generatedPath} and commit the change`
-  )
+try {
+  execa.sync('git', ['--version'])
+  if (execa.sync('git', ['diff-index', '--name-only', 'HEAD']).stdout.includes(rawCompilerOptionsFileName)) {
+    throw new Error(
+      `Tsconfig options have changed. Please check the modified generated ${generatedPath} and commit the change`
+    )
+  }
+} catch (e) {
+  console.log('git command is not available. Skip checking generated types')
 }
-
 computePackageDigest()

